src/Altinn.AccessManagement.Core/Services/DelegationsService.cs

```diff
@@ -224,6 +224,86 @@ public async Task<List<Delegation>> GetMaskinportenSchemaDelegations(string supp
             return await GetAllMaskinportenSchemaDelegations(supplierPartyId, consumerPartyId, scope);
         }
 
+        /// <inheritdoc/>
+        public async Task<DelegationOutput> RevokeMaskinportenDelegation(int authenticatedUserId, DelegationInput delegation)
+        {
+            (DelegationOutput output, string resourceRegistryId, Party fromParty, Party toParty) = await ValidateMaskinportenDelegationModel(delegation);
+            if (!output.IsValid)
+            {
+                return output;
+            }
+
+            List<RequestToDelete> policiesToDelete = DelegationHelper.GetRequestToDeleteResourceRegistryService(authenticatedUserId, resourceRegistryId, fromParty.PartyId, toParty.PartyId);
+
+            await _pap.TryDeleteDelegationPolicies(policiesToDelete);
+            return output;
+        }
+
+        private async Task<(DelegationOutput Output, string ResourceRegistryId, Party FromParty, Party ToParty)> ValidateMaskinportenDelegationModel(DelegationInput delegation)
```

Comment:
The error messages either need to be updated to be correctly referring to revoke and not delegation, or be made generic. to be used in both contexts? e.g.:
"Maskinporten schema delegation must specify..." -> "This operation requires ..."?

Reply:
Both delegation and revoke now use the same validation method.